scripts/release/createNativeModules.js

```diff
@@ -0,0 +1,203 @@
+    const jsActions = await getFiles(jsActionsPath);
+    const tmpFolderWidgets = join(tmpFolder, "widgets");
+    const tmpFolderActions = join(tmpFolder, "javascriptsource/nativemobileresources/actions");
+
+    console.log("Updating NativeComponentsTestProject..");
+    const githubUrlDomain = githubUrl.replace("https://", "");
+    const githubUrlAuthenticated = `https://${process.env.GH_USERNAME}:${process.env.GH_PAT}@${githubUrlDomain}`;
+    await rm(tmpFolder, { recursive: true, force: true });
+    await execShellCommand(`git clone ${githubUrlAuthenticated} ${tmpFolder}`);
+
+    console.log("Copying widgets and js actions..");
+    await Promise.all([
+        ...nativeWidgetFolders.map(async folder => {
+            const src = (await getFiles(folder, [`.mpk`]))[0];
+            const dest = join(tmpFolderWidgets, basename(src));
+            await rm(dest);
+            await copyFile(src, dest);
+        }),
+        ...jsActions.map(async file => {
+            const dest = join(tmpFolderActions, file.replace(jsActionsPath, ""));
+            await rm(dest);
+            await copyFile(file, dest);
+        })
+    ]);
+    const gitOutput = await execShellCommand(`cd ${tmpFolder} && git status`);
+    if (!/nothing to commit/i.test(gitOutput)) {
+        await execShellCommand(
+            `cd ${tmpFolder} && git add . && git commit -m "Updated native widgets and js actions" && git push`
+        );
+    } else {
+        console.warn(`Nothing to commit from repo ${tmpFolder}`);
+    }
+}
+
+// Create reusable mxbuild image
+async function createMxBuildContainer(sourceDir, moduleName, mendixVersion) {
+    const existingImages = (await execShellCommand(`docker image ls -q mxbuild:${mendixVersion}`)).toString().trim();
+    if (!existingImages) {
+        console.log(`Creating new mxbuild docker image...`);
+        await execShellCommand(
+            `docker build -f ${join(
+                process.cwd(),
+                "packages/tools/pluggable-widgets-tools/scripts/mxbuild.Dockerfile"
+            )} ` +
+                `--build-arg MENDIX_VERSION=${mendixVersion} ` +
+                `-t mxbuild:${mendixVersion} ${process.cwd()}`
+        );
+    }
+
+    // Build testProject via mxbuild
+    const projectFile = basename((await getFiles(sourceDir, [`.mpr`]))[0]);
+    await execShellCommand(
+        `docker run -t -v ${sourceDir}:/source ` +
+            `--rm mxbuild:${mendixVersion} bash -c "mx update-widgets --loose-version-check /source/${projectFile} && mono /tmp/mxbuild/modeler/mxutil.exe create-module-package /source/${projectFile} ${moduleName}"`
+    );
+    console.log(`Module ${moduleName} created successfully.`);
+}
+
+function execShellCommand(cmd) {
+    return new Promise((resolve, reject) => {
+        const cwd = process.cwd();
+        exec(cmd, { cwd }, (error, stdout, stderr) => {
+            if (error) {
+                console.warn(stderr);
```

Comment:
You guard then log this below, line 184. This can be removed

Reply:
Sometimes a command outputs stderr while it's actually a warning. We only want to reject the promise when the command fails completely, therefore I think we need both checks :)

Reply:
okay 👍